### PR TITLE
Move Terraform CI/CD to GitHub Actions 

### LINKED ISSUE #1211: Move Terraform CircleCI jobs to GitHub Actions
**Motivation**

In order to move CI/CD pipeline as much as possible to GitHub Actions. 

**Feature**

Configure GitHub Actions to run the current cloud infrastructure CI/CD pipeline.
Cleanup the pipeline configured for CircleCI.

**Alternatives**

NA

**Additional context**

NA

### PR DESCRIPTION
**Pull request by maxgio92**
This PR moves the CICD pipeline of the cloud infrastructure managed with Terraform, from CircleCI to GitHub.

Fixes #1211

### CODE REVIEW

File: .github/workflows/terraform-apply.yml
Comment (maxgio92) [inline .github/workflows/terraform-apply.yml:41]: This is not how should be done, as the desired state (the TF plan) is re-calculated, during this workflow, which is triggered by new commits on the master branch, when changes affect the infrastructure-code.
Documentation: https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform#interactive-approval-of-plans.

Nevertheless, I think it's a valid trade-off, considering that this specific GitHub event doesn't link to a PR AFAIK, and it would be hard to withdraw the correct plan which was generated in the respective approved PR.

**Comment by maxgio92** [comment]
I'm sorry I didnt' consider #961. Anyway it cleans up the wrapper bash scripts and it uses Terraform gihub actions directly.

Furthermore, it executes the Terraform plan during PR submission in order to be able to verify the expected plan apply, before merging it.

Anyway, this PR needs rebase on top of #961

**Comment by maxgio92** [comment]
Sure @FedeDP

File: .github/workflows/terraform-apply.yml
Comment (FedeDP) [reply .github/workflows/terraform-apply.yml:41]: The current master gha leverages the `concurrency` keyword: https://github.com/falcosecurity/test-infra/blob/master/.github/workflows/master.yml#L7.
This allows us to enqueue subsequent master CI jobs until each one of them finishes (in a FIFO manner).

File: .github/workflows/terraform-apply.yml
Comment (FedeDP) [inline .github/workflows/terraform-apply.yml:14]: I think we should just configure aws credentials like we do in master CI: https://github.com/falcosecurity/test-infra/blob/master/.github/workflows/master.yml#L21

File: .github/workflows/terraform-plan.yml
Comment (FedeDP) [inline .github/workflows/terraform-plan.yml:17]: I don't love for a PR CI to have write access to the cluster (using eg: https://github.com/falcosecurity/test-infra/blob/master/.github/workflows/master.yml#L21).
Do we really need this?

File: .github/workflows/terraform-apply.yml
Comment (maxgio92) [reply .github/workflows/terraform-apply.yml:14]: I totally agree to follow the best practice to assume a role. Thanks @FedeDP

File: .github/workflows/terraform-plan.yml
Comment (maxgio92) [reply .github/workflows/terraform-plan.yml:17]: Write actually is not needed, we should setup a tailored IAM Role.
The authentication instead is needed to generate the plan

File: .github/workflows/terraform-apply.yml
Comment (FedeDP) [reply .github/workflows/terraform-apply.yml:41]: We are not running them concurrently, we are using the `concurrency` group to create a "queue" of master deployments, so that indeed we never run them concurrntly.

Comment (maxgio92) [reply .github/workflows/terraform-apply.yml:41]: > We are not running them concurrently, we are using the `concurrency` group to create a "queue" of master deployments, so that indeed we never run them concurrntly.

Oh ok thanks, I misunderstood about what would be queued.
Then I agree, I'm going to update it.
In addition to this barrier, theres a write lock during apply, so we're safe I guess :)

Comment (maxgio92) [reply .github/workflows/terraform-apply.yml:48]: Added @FedeDP, feel free to resolve this conversation

File: .github/workflows/terraform-apply.yml
Comment (maxgio92) [reply .github/workflows/terraform-apply.yml]: Implemented @FedeDP

File: config/clusters/prow.auto.tfvars
Comment (maxgio92) [inline config/clusters/prow.auto.tfvars:26]: PTAL there: https://github.com/falcosecurity/test-infra/pull/1212

File: .github/workflows/master.yml
Comment (FedeDP) [inline .github/workflows/master.yml:27]: Also, we can cleanup the script, right?
Moreover, i'd rename this CI to `Master CI - deploy prow` and the new one to `Master CI - apply terraform`.

Comment (maxgio92) [reply .github/workflows/master.yml:27]: Totally agree @FedeDP. Implemented, PTAL

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**
